django/core/files/uploadedfile.py

```diff
@@ -58,7 +58,8 @@ class TemporaryUploadedFile(UploadedFile):
     A file uploaded to a temporary location (i.e. stream-to-disk).
     """
     def __init__(self, name, content_type, size, charset, content_type_extra=None):
-        file = tempfile.NamedTemporaryFile(suffix='.upload', dir=settings.FILE_UPLOAD_TEMP_DIR)
+        ext = os.path.splitext(name)[1]
```

Comment:
I think: `_, ext = os.path.splitext(name)` is a bit more readable than "magical" indexing.